test/client.test.js

```diff
@@ -2376,6 +2376,65 @@ describe('ACC Client', function () {
             client._transport.mockReturnValueOnce(Mock.GET_NMS_EXTACCOUNT_SCHEMA_RESPONSE);
             await client.getSchema("nms:extAccount");
         })
+
+        it("Should clear storage if necessary", async () => {
+            const sdkVersion = sdk.getSDKVersion().version;
+            const mockSdkVersionItemKey    = "acc.js.sdk.0.0.0.acc-sdk:8080.cache.Hello";
+            const otherItemKey             = "other.Hello";
+            const map = {};
+            const storage = {
+                "acc.js.sdk.0.0.0.acc-sdk:8080.cache.Hello": "0.0.0.World",
+                "other.Hello": "other.World",
+                getItem: jest.fn((key) => map[key]),
+                setItem: jest.fn((key, value) => map[key] = value),
+                removeItem: jest.fn((key) => { delete map[key] })
+            }
+
+            let client = await Mock.makeClient({ storage: storage });
+
+            // Remove item not matches to the current sdk version and begins with "acc.js.sdk"
+            expect(storage.removeItem.mock.calls.length).toBe(1);
+            expect(storage.removeItem.mock.calls[0][0]).toMatch(mockSdkVersionItemKey);
+
+            client._transport.mockReturnValueOnce(Mock.LOGON_RESPONSE);
+            await client.NLWS.xtkSession.logon();
+            client._transport.mockReturnValueOnce(Mock.GET_NMS_EXTACCOUNT_SCHEMA_RESPONSE);
+            await client.getSchema("nms:extAccount");
+            const currentSdkVersionItemKey = "acc.js.sdk."+sdkVersion+".acc-sdk:8080.cache.XtkEntityCache$xtk:schema|nms:extAccount";
+            expect(storage.setItem.mock.calls[0][0]).toMatch(currentSdkVersionItemKey);
+
+            // Remove ONLY the item not matches to the current sdk version but not the one matches
+            // Now simulate reusing the local storage to make sure that we remove ONLY the item 
+            // not matches to the current sdk version.
+            // We need a new client to make sure we do not reuse the in-memory cache of the client.
+            client = await Mock.makeClient({ storage: storage });
+            client._transport.mockReturnValueOnce(Mock.LOGON_RESPONSE);
+            await client.NLWS.xtkSession.logon();
+            client._transport.mockReturnValueOnce(Mock.GET_NMS_EXTACCOUNT_SCHEMA_RESPONSE);
+            await client.getSchema("nms:extAccount");
```

Comment:
There is no `expect()` , what are you testing here ?

Reply:
modified